Modules/Core/Common/include/itkBSplineInterpolationWeightFunction.hxx

```diff
@@ -18,14 +18,19 @@
 #ifndef itkBSplineInterpolationWeightFunction_hxx
 #define itkBSplineInterpolationWeightFunction_hxx
 
-#include "itkBSplineInterpolationWeightFunction.h"
```

Comment:
Why should `#include "itkBSplineInterpolationWeightFunction.h"` be removed from the corresponding hxx file?

Reply:
I'll deal with this more uniformly in a different commit.  The issues is IDE warning based.  The recursive nature of the .h file including the .hxx file which in turn includes the .h file again causes the IDE to issue warnings about unterminated @ifdef's 

I don't see why the recurse calls should be there unless you want to #include the .hxx instead of the .h file, which I don't see as necessary.

Reply:
@hjmjohnson Wasn't the #include "itk....h" there in the hxx file in order to support `ITK_MANUAL_INSTANTIATION`? So that users could have a dedicated cxx file that would only #include hxx files, and then manually instantiate what they need?  I'm just asking.

Reply:
I think that was the theory back in 2001, but I don't think that use case is in practice.  In the possible very rare case that it was used that way, we could simply include the .h file and the .hxx file in the cxx file.